File: src/main/java/dailyChallege/Intersect.java
package dailyChallege;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Intersect {
	public int[] intersect(int[] nums1, int[] nums2) {

		Map<Integer, Integer> map = new HashMap<>();
		for(int i : nums1) {
			map.put(i, map.getOrDefault(i, 0)+1);
		}
		List<Integer> list = new ArrayList<>();
		for(int i : nums2) {
			if(map.containsKey(i) && map.get(i)>0) {
				list.add(i);
				map.put(i, map.get(i)-1);
			}
		}
		int[] res = new int[list.size()];
		for(int i=0;i<res.length;i++) {
			res[i]=list.get(i);
		}
		return res;

	}

}

File: python/intersect.py
class Solution(object):
    def intersect(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        map1 = {}
        res=[]
        for i in nums1:
            map1[i] = map1.get(i,0)+1
        for i in nums2:
            if i in map1 and map1[i]>0:
                res.append(i)
                map1[i]-=1
        return res